### ISSUE TITLE
[2.0] MKS Gen L - layer fan not working with FAN_SOFT_PWM (MOSFET LED is barely lit) - works with 1.1.8

### ISSUE THREAD

**Issue by dayaan82** [MarlinFirmware/Marlin#13389]
Almost explained it all in the description.

FAN_SOFT_PWM enabled.. layer fan not working anymore.. the MOSFET LED is barely lit and not bright like the extruder and bed ones..

Works normally with FAN_SOFT_PWM disabled.

This option works on bugfix 1.1.x.

**Comment by ghost** [comment]
Yes, I have exactly the same problem. Board = MKS Gen 1.4, with Marlin 2.0.x from September 2018,  which I have on my hard disk SOFT_PWM works fine. With Marlin 2.0.x downloaded yesterday - SOFT_PWM does not work. All settings in Configuration.h and Configuration_adv.h are the same.

**Comment by thisiskeithb** [comment]
A friend of mine was having similar issues with their MKS GEN 1.4 board and a [24v Mechatronics B5015E24B 5015 radial fan](https://www.digikey.com/product-detail/en/mechatronics-fan-group/B5015E24B-BSR/1570-1034-ND/5209731). I purchased the exact same fan and confirmed the same issue using Marlin 1.1.9 & the latest Bugfix-2.0 branch from the [1440187](https://github.com/MarlinFirmware/Marlin/commit/144018736aa1e0d466e0c0693cf7992190e4981f) commit from a couple of days ago.

I had to enable` FAST_PWM_FAN` and disable `FAN_SOFT_PWM`, otherwise the fan wouldn't spin at any value between 1 & 255. It would just make a faint clicking sound and the mosfet LED would barely light up.

Edit: This issue does not occur with the lower-end 5015 SoundOriginal fans commonly found online.

**Comment by bossman74** [comment]
I have the same issue with this 12v Brushless 5015 blower.  But no issues when using Marlin 2.0.x from last August, no noise at all speeds.  This looks like a recent thing.

On the latest branch, I only got it running with `FAN_SOFT_PWM` commented.  And then, it's noisy when not running at full speed.

[https://goo.gl/ePVwPx](url)

**Comment by bossman74** [comment]
Anyone found a solution to this ?  I reverted back to an older 2.0.x bugfix until then.

**Comment by ghost** [comment]
For all of you who would like to use the latest version of Marlin (where SOFT_PWM does not work) but have a problem with a loud fan.

I used simple hardware in the form of a low-pass filter. After connecting this you can turn off SOFT_PWM (use Hardware PWM) and the whistling noise is gone.
![Zrzut ekranu 2019-03-27 o 08 02 52](https://user-images.githubusercontent.com/48625824/55056698-28ed3a80-5067-11e9-924c-cf7c91965a59.png)

**Comment by bossman74** [comment]
In the latest commit from yesterday, my fan now works with `FAN_SOFT_PWM` and without the noise.

Thank you to whoever fixed it !

**Comment by jasonmarquette** [comment]
I have an issue with my parts fan not running after upgrading to marlin 2.0.3. I also have an MKS_GEN_L board. I'm using a BLTouch 3.1

My Fan will not turn on via any slicer settings, however I can control it on the Ender 3 directly and it will turn on manually. I was really hoping enabling FAN_SOFT_PWM would solve my issue. I have searched everywhere. Any Idea's? 

UPDATED:
This is not an issue. I didn't have the correct pin defined for my auto_fan. For the MKS GEN L it is pin 7. I did this, and it fixed it:
`#define E0_AUTO_FAN_PIN 7`
